crates/cluster_agent/src/log_metadata/log_metadata_watcher.rs

```diff
@@ -706,4 +706,359 @@ mod test {
             );
         }
     }
+
+    #[tokio::test]
+    #[cfg(not(target_os = "macos"))]
+    #[parallel]
+    async fn test_newly_created_file_receives_modify_events() {
+        let temp_dir = tempfile::tempdir().expect("Failed to create temp dir");
+        let namespaces = vec!["modify-namespace".into()];
+        let (term_tx, _term_rx) = broadcast::channel(1);
+        let logs_dir = temp_dir.path().to_owned();
+
+        let (log_metadata_watcher, mut log_metadata_rx) = LogMetadataWatcher::new(
+            logs_dir.clone(),
+            namespaces,
+            term_tx.clone(),
+            "The node name".to_owned(),
+        );
+
+        task::spawn(async move { log_metadata_watcher.watch::<RecommendedWatcher>(None).await });
+
+        while term_tx.receiver_count() != 2 {
+            sleep(Duration::from_millis(50)).await;
+        }
+
+        sleep(Duration::from_millis(200)).await;
+
+        let file_path =
+            logs_dir.join("pod-name_modify-namespace_container-name-newcontainerid.log");
+        let mut file = File::create(&file_path).expect("Failed to create file");
+        file.write_all(&vec![0; 4])
+            .expect("Failed to write initial data");
+        file.flush().expect("Failed to flush");
+
+        sleep(Duration::from_millis(500)).await;
+
+        file.write_all(&vec![1; 3])
+            .expect("Failed to write additional data");
+        file.flush().expect("Failed to flush");
+        drop(file);
+
+        sleep(Duration::from_millis(500)).await;
+
+        let mut events = Vec::new();
+        let timeout_duration = Duration::from_secs(5);
```

Comment:
Can we run the entire test suite under a timeout rather than handling the timeouts explicitly in the test code?